### ISSUE TITLE
Rule Request: Omit string raw value enum declarations identical to the its identifiers

### ISSUE THREAD

**Issue by iliaskarim** [realm/SwiftLint#2830]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### New rule request

Please describe the rule idea, format
this issue's title as `Rule Request: [Rule Name]` and describe:

1. Why should this rule be added? Share links to existing discussion about what
   the community thinks about this.  

This rule could help Swift-adopters learn about the raw-value generating string raw-value enum.

2. Provide several examples of what _would_ and _wouldn't_ trigger violations.

__Triggering__
```
enum AnEnum: String {
  case one = "one"
  case two = "two"
}
```

__Not-triggering__
```
enum AnEnum: String {
  case one
  case two
}

enum AnotherOne: String {
  case one = "One"
  case two = "Two"
}

enum AnotherEnum: Int {
  case one = 0
  case two = 1
}
```

3. Should the rule be configurable, if so what parameters should be configurable?

No the rule should not be configurable.

4. Should the rule be opt-in or enabled by default? Why?

The rule should be enabled by default because it's best practice for readability.

**Comment by marcelofabri** [comment]
`redundant_string_enum_value` should catch that: https://github.com/realm/SwiftLint/blob/master/Rules.md#redundant-string-enum-value
